docs/concepts.md

```diff
@@ -0,0 +1,159 @@
+
+There are five profiles on sugoi permission model : admin, read and write, password and application manager. A user have a right when it belong to the corresonding ldap group. All profiles are scoped to realm except the admin group. Sugoi management groups are entirely configurable with regex.
+
+### Administrator
+
+The admin group is defined via the regexp.role.admin property.
+
+An administrator can :
+
+* read, create, modify or delete any realms.
+* read, create, modify or delete userstorages.
+* do whatever the writer can do on all realms.
+
+### Writer
+
+The writer group is defined via the regexp.role.writer property. It is scoped to one realm.
+
+A writer can :
+
+* modify and create users on the realms they manage
+* modify and create organizations on the realms they manage
+* modify and create application and groups on the realms they manage
+* do whatever a reader can do on the realms they manage.
+
+### Reader
+
+The reader group is defined via the regexp.role.reader property. It is scoped to one realm.
+
+A reader can :
+
+* read users, organizations, applications and groups on the realm they are reader on.
+
+### Password manager
+
+The password manager is defined via the regexp.role.password.manager property. It is scoped to one realm.
+
+Password manager can initialize passwords, update password (with a given password or not) or validate a password on its realm.
+
+### Application manager
+
+Application manager can manage an application and the associated groups. This also enable reader rights on a realm.
+
+## Providers
+
+### Realm config provider
+
+#### File
+
+This provider loads *realms* from a file described in `fr.insee.sugoi.realm.config.local.path`. The file is fetched through Spring [ResourceLoader](https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#resources-resourceloader).
+
+The file is expected to contain an array of realm representations, like : 
+```json
+[
+    {
+        "name": "test",
+        "url": "/test",
+        "appSource": "/data/applications",
+        "userStorages": [
+            {
+                "name": "default",
+                "userSource": "/data/users",
+                "organizationSource": "/data/organizations",
+                "properties": null,
+                "readerType": null,
+                "writerType": null
+            }
+        ]
+    }
+]
+```
+
+#### Ldap
+
+This provider loads realm from ldap entries, an exemple ldif file and configuration can be found in the [sugoi-distribution-full-env](../sugoi-api-distribution/sugoi-api-distribution-full-env/src/main/resources) module.
+
+
+
+
+### Store Providers
+
+They are responsible to effectively read and write sugoi objects (users, organizations and applications) in a store.
+
+
+#### File Store Provider
+
+Each object is stored in a json file located in the folder located as configured in the realm.
```

Comment:
in the folder as configured in the realm instead of located as configured in the realm ?